ndcube/utils/wcs.py

```diff
@@ -197,18 +201,31 @@ def _wcs_slicer(wcs, missing_axis, item):
                 item_checked.append(slice(0, 1))
         # if all are slice in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
             for i, it in enumerate(item_checked):
                 if isinstance(it, int):
                     missing_axis[i] = True
     # returning the reverse list of missing axis as in the item here was reverse of
     # what was inputed so we had a reverse missing_axis.
-    return new_wcs, missing_axis[::-1]
+    extra_coords_list = [] # initiating new list
+    pix_coords = [0] * len(item_) # Set up a list of pixel coords as input to all_pix2world
+    # Checking item_ slices for dropped axes if any
+    for i, slice_element in enumerate(item_):
+        if missing_axis[i] is False:
+            if (slice_element.stop - slice_element.start) == 1:
+                pix_coords[i] = slice_element.start 
+                # Enter pixel coordinate for this axis. 
+                # Since we are only dealing with independent axes the other axes can remain at 0.
+                real_world_coords = wcs.all_pix2world(tuple(pix_coords), 0)
+                axis_name = slice_element.wcs.ctype
```

Comment:
I am not sure about this bit... I tried something similar on my machine and it kind of worked. I think my understand about what a slice is and how it works is still a work-in-progress. 
```
axis_name = slice_element.wcs.ctype
```

Reply:
A ```slice``` (named here as ```slice_element```) is simply an object that hold the ```start``` and ```stop``` indices of an interval.  It also holds the ```step``` parameter, which enables you to access every 2nd, 3rd, 4th, etc. element within that interval.  The default ```step``` is 1.

A ```slice``` object can be applied to any iterable, e.g. strings, lists, tuple, arrays, etc. to extract a desired interval within it.  It is a more explicit way of storing the slice information than the standard indexing notation, e.g.
```python
>>> my_list = [11, 12, 13, 14]
>>> my_list[1:3]
[12, 13]
>>> my_list[slice(1,3)]
[12, 13]
```

For more info, see this tutorial: https://www.programiz.com/python-programming/methods/built-in/slice

Therefore, in our case, we construct a list or tuple of ```slice``` items, one for each axis, which will be used to slice the ```wcs``` object.  But we can use the info in those ```slice``` objects, i.e. whether the ```stop - start == 1```, to know whether an axis will be dropped from the old ```wcs``` object when creating the ```new_wcs``` object.  Then we can find out what the real world coordinate at the position along the dropped axis by using the old ```wcs.all_pix2world``` and the index information from the ```slice``` object.  Let me know if this makes sense.  But do check out that tutorial first as it may help with understanding this last paragraph.

Reply:
Yup, it makes sense. So I reckon the `slice` object is kind of like a list (or a cut) of indices for us to help keep track of and manipulate the iterable concerned...